src/class-wc-gateway-btcpay.php

```diff
@@ -973,14 +984,9 @@ public function ipn_callback()
                 $this->log('    [Info] Order ID is: ' . $order_id);
```

Comment:
Maybe also add here the order number like on the other logs or not interesting in this case? 

What is a bit problematic above in Line 977 imo (nothing you changed but let's talk about it?):
`$order_id = $invoice->getOrderId();`

So we override the original variable by reading our $invoice object order id. I think we should not override the $order_id here as it could be completely different from the invoice order id for various reasons. To be safe here I would rather be explicit and use $invoice_order_id here. Thoughts?

Thinkng about this more, this was also likely the bug you fixed with this PR. The code before your change was `$invoice->setOrderId((string)$order_number);`. So display order number was set to be the invoice order id. And here line 977 the internal $order_id gets overridden with $order_number for all the code that follows below. 😬

Reply:
> What is a bit problematic above in Line 977 imo (nothing you changed but let's talk about it?): $order_id = $invoice->getOrderId();

This is in the `ipn_callback()` function – it doesn't override the variable, but initializes it for the first time in this function.

> this was also likely the bug you fixed with this PR.

Yes, in general we are now using the order ID consistently on the way out and way in, so there is no ambiguity.

Reply:
Ahh, got it, I did expand too few lines and assumed it is in the huge process_payment function, sorry. So yes correct, in this case getting the $order_id from the $invoice is correct and the only way. Insofar your fix is very important to make sure the internal order id get's set for the invoice and not the altered one.